ouroboros-consensus-test/src/Test/Util/Serialisation/Roundtrip.hs

```diff
@@ -231,14 +231,29 @@ instance ( blockVersion ~ BlockNodeToClientVersion blk
       Query.TopLevelQueryDisabled ->
         arbitraryBlockQuery queryVersion
 
-      Query.QueryVersion1 ->
+      Query.QueryVersion1 -> genTopLevelQuery queryVersion
+      Query.QueryVersion2 -> genTopLevelQuery queryVersion
+    where
+      genTopLevelQuery queryVersion =
         frequency
-          [ (15, arbitraryBlockQuery queryVersion)
-          , (1,  do blockV <- arbitrary
-                    return (WithVersion (queryVersion, blockV)
-                                        (SomeSecond GetSystemStart)))
+          [ queryFrequencyFor Query.TopLevelQueryDisabled 15 $
```

Comment:
One thing that I'd like to discuss in a call is why not just hardcoding the frequencies as it used to be

Reply:
So, in particular, I was wondering if it'd be possible to write something like:

```haskell
    case queryVersion of
      Query.TopLevelQueryDisabled -> arbitraryBlockQuery queryVersion
      Query.QueryVersion1         -> arbitraryBlockVersion1Query
      Query.QueryVersion2         -> arbitraryBlockVersion1Query

  where
    query item = do blockV <- arbitrary
                     return (WithVersion (queryVersion, blockV)
                                         (SomeSecond item)))
    arbitraryBlockVersion1Query =
      frequency
        [ (15, arbitraryBlockQuery queryVersion)
        , (1,  query GetSystemStart)))
        ]
    arbitraryBlockVersion2Query =
      oneof
        [ arbitraryBlockVersion1Query
        , frequency [ (1, query GetChainBlockNo)
                    , (1, query GetChainPoint)
                    ]
        ]
```

Reply:
Or, repeating code, but removing one level of indirection for the reader so that it is easier for them to understand what the frequency distributions are:

```haskell
    arbitraryBlockVersion2Query =
      frequency
        [ (15, arbitraryBlockQuery queryVersion)
        , (1,  query GetSystemStart)))
        , (1, query GetChainBlockNo)
        , (1, query GetChainPoint)
        ]
```

Reply:
How does it look now?